### PR TITLE
StatefulSet: Remove `pod.alpha.kubernetes.io/initialized` annotation.

### PR DESCRIPTION
**Pull request by enisoc**
The `pod.alpha.kubernetes.io/initialized` annotation was originally a tool for validating StatefulSet's ordered Pod creation guarantees during the feature's alpha phase.

If set to "false" on a given Pod, it would interrupt StatefulSet's normal behavior. In v1.5.0, the annotation was deprecated and the default became "true" as part of StatefulSet's graduation to beta.

The annotation is now ignored, meaning it cannot be used to interrupt StatefulSet Pod management.

```release-note
StatefulSet: The deprecated `pod.alpha.kubernetes.io/initialized` annotation for interrupting StatefulSet Pod management is now ignored. If you were setting it to `true` or leaving it unset, no action is required. However, if you were setting it to `false`, be aware that previously-dormant StatefulSets may become active after upgrading.
```

ref #41605

### CODE REVIEW

**Review by kow3ns — APPROVED** [review]
This approach is exactly what I had in mind as well. Pausing the StatefulSet, for testing purposes, by breaking its readiness probe achieves the exact same result as the removed annotation without unnecessarily exposing implementation details, and will meet the requirements of backward compatibility.

1. Do you intend to cherry pick b8214b235d135c92264ce69145498a10cfaef1bb into 1.7 e2e so that upgrade tests will function?
1. Also we should consider removing the annotation from the v1beta2 API.

Both of the above can be addressed in future PRs.

**Comment by enisoc** [comment]
@kow3ns Good points. I've added TODO items to https://github.com/kubernetes/kubernetes/issues/41605#issuecomment-317058079.

**Comment by enisoc** [comment]
I'm waiting for #49400 before assigning for approval.

**Pull request merged**
